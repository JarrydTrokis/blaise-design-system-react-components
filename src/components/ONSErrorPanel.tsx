import React, {ReactElement} from "react";

/**
 * This is a Standard Panel with an error status, with a prepared message.
 */
function ONSErrorPanel(): ReactElement {
    return (
        <>
            <div className="ons-panel ons-panel--error ons-panel--no-title">
                <div className="ons-panel__body">
                    <p>
                        Sorry, there is a problem with this service. We are working to fix the problem. Please try again later.
                    </p>
                </div>
            </div>
        </>
    );
}

export default ONSErrorPanel;
