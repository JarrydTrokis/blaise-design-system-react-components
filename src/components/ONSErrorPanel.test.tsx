import { render } from "@testing-library/react";
import React from "react";
import ONSErrorPanel from "./ONSErrorPanel";

describe("ONS Error Panel Test", () => {
    it("matches Snapshot", () => {
        expect(render(<ONSErrorPanel />)).toMatchSnapshot();
    });

    it("should render correctly", () => {
        const { container } = render(<ONSErrorPanel />);
        expect(container).toBeDefined();
    });
});
