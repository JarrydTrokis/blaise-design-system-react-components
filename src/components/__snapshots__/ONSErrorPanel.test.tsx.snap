// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS Error Panel Test matches Snapshot 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "ons-panel ons-panel--error ons-panel--no-title",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "ons-panel__body",
        },
        "children": Array [
          Node {
            "attribs": Object {},
            "children": Array [
              Node {
                "data": "Sorry, there is a problem with this service. We are working to fix the problem. Please try again later.",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "p",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
